src/System.Management.Automation/CoreCLR/CorePsPlatform.cs

```diff
@@ -554,27 +554,289 @@ internal static int NonWindowsGetProcessParentPid(int pid)
+                OtherWrite     = 0x002,
+                
+                /// <summary>The mask to get the "other" execute bit.</summary>
+                OtherExecute   = 0x001,
+
+                /// <summary>The mask to retrieve the sticky bit.</summary>
+                SetStickyMask  = 0x200,
+                
+                /// <summary>The mask to retrieve the setgid bit.</summary>
+                SetGidMask     = 0x400,
+
+                /// <summary>The mask to retrieve the setuid bit.</summary>
+                SetUidMask     = 0x800,
+            }
+
+            /// <summary>The Common Stat class.</summary>
+            public class CommonStat
+            {
+                /// <summary>The inode of the filesystem item.</summary>
+                public long Inode;
+
+                /// <summary>The Mode of the filesystem item.</summary>
+                public int Mode;
+
+                /// <summary>The user id of the filesystem item.</summary>
+                public int UserId;
+
+                /// <summary>The group id of the filesystem item.</summary>
+                public int GroupId;
+
+                /// <summary>The number of hard links for the filesystem item.</summary>
+                public int HardlinkCount;
+
+                /// <summary>The size in bytes of the filesystem item.</summary>
+                public long Size;
+
+                /// <summary>The last access time of the filesystem item.</summary>
+                public DateTime AccessTime;
+
+                /// <summary>The last modified time for the filesystem item.</summary>
+                public DateTime ModifiedTime;
+
+                /// <summary>The last time the status changes for the filesystem item.</summary>
+                public DateTime StatusChangeTime;
+
+                /// <summary>The block size of the filesystem.</summary>
+                public long BlockSize;
+
+                /// <summary>The device id of the filesystem item.</summary>
+                public int DeviceId;
+
+                /// <summary>The number of blocks used by the filesystem item.</summary>
+                public int NumberOfBlocks;
+
+                /// <summary>The type of the filesystem item.</summary>
+                public ItemType ItemType;
+
+                /// <summary>Whether the filesystem item has the setuid bit enabled.</summary>
+                public bool IsSetUid;
+
+                /// <summary>Whether the filesystem item has the setgid bit enabled.</summary>
+                public bool IsSetGid;
+
+                /// <summary>Whether the filesystem item has the sticky bit enabled. This is only available for directories.</summary>
+                public bool IsSticky;
+
+                // helper for getting unix mode
+                private Dictionary<StatMask, string> modeMap = new Dictionary<StatMask, string>()
+                {
+                        { StatMask.OwnerRead, "r" },
```

Comment:
Maybe a slight optimization, if you declare "r", "w" and "x" as constants, then they can be reused.

Reply:
done!